src/BaseClient.php

```diff
@@ -108,53 +136,137 @@ public function getToken(): ?array
      * @throws Exception when something unexpected went wrong.
      */
     public function authenticate(string $clientId, string $clientSecret): void
+    {
+        $tokenRequest = TokenRequest::constructFromArray([
+            'grant_type' => 'client_credentials',
+        ]);
+
+        $this->requestToken($clientId, $clientSecret, $tokenRequest, 'RETAILER');
+    }
+
+    /**
+     * Authenticates with Bol.com Retailer API Server using an authorization code.
+     *
+     * @param string $clientId The client ID to use for authentication.
+     * @param string $clientSecret The client secret to use for authentication.
+     * @param string $code Authorization code received from Bol.com.
+     * @param string $redirectUri The redirect URI used for the authorization code request.
+     * @return JWTToken The refresh token.
+     *
+     * @throws ConnectException
+     * @throws Exception
+     * @throws RateLimitException
+     * @throws ResponseException
+     * @throws UnauthorizedException
+     */
+    public function authenticateByAuthorizationCode(string $clientId, string $clientSecret, string $code, string $redirectUri): JWTToken
+    {
+        $tokenRequest = TokenRequest::constructFromArray([
+            'grant_type' => 'authorization_code',
+            'code' => $code,
+            'redirect_uri' => $redirectUri,
+        ]);
+
+        $tokenResponse = $this->requestToken($clientId, $clientSecret, $tokenRequest);
+
+        return new JWTToken($tokenResponse->refresh_token);
+    }
+
+    /**
+     * Refreshes the access token and refresh token (latter depending on your account settings). This requires an
+     * existing token with a refresh token.
+     *
+     * @param string $clientId The client ID to use for authentication.
+     * @param string $clientSecret The client secret to use for authentication.
+     * @param JWTToken $refreshToken The refresh token.
+     * @return JWTToken The (current or new) refresh token, depending on the authentication settings managed by Bol.com
+     * for your account.
+     *
+     * @throws ConnectException
+     * @throws Exception
+     * @throws RateLimitException
+     * @throws ResponseException
+     * @throws UnauthorizedException
+     */
+    public function refreshAccessToken(string $clientId, string $clientSecret, JWTToken $refreshToken): JWTToken
```

Comment:
Authenticatie met een refresh token is (binnen OAuth 2.0) gewoon een andere Grant type, dus deze methode zou je ook `authenticateByRefreshToken` kunnen noemen, dan is de naamgeving van de `authenticate*` methodes mooi consistent 🙂